File: nomad-samples/admin/py/basic-live-channel-1.1/live_channel/create_live_channel.py
from constants.project_constants import *
from exceptions.api_exception_handler import *
from helpers.slugify import *
from live_channel.live_channel_statuses import *
from live_channel.wait_live_channel_status import *
from live_channel.live_channel_types import *

import json, requests

def create_live_channel(AUTH_TOKEN, NAME, THUMBNAIL_IMAGE, ARCHIVE_FOLDER_ASSET, 
                        ENABLE_HIGH_AVAILABILITY, ENABLE_LIVE_CLIPPING, IS_SECURE_OUTPUT, 
                        OUTPUT_SCREENSHOTS, TYPE, URL):
    # Create header for the request
    HEADERS = {
        'Content-Type': 'application/json',
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    # Build the payload BODY
    BODY = {
        "name": NAME,
        "routeName": slugify(NAME),
        "enableHighAvailability": ENABLE_HIGH_AVAILABILITY,
        "enableLiveClipping": ENABLE_LIVE_CLIPPING,
        "isSecureOutput": IS_SECURE_OUTPUT,
        "outputScreenshots": OUTPUT_SCREENSHOTS,
        "type": { "id": LIVE_CHANNEL_TYPES[TYPE] }
    }

    if THUMBNAIL_IMAGE != "":
        BODY["thumbnailImage"] = { "id": THUMBNAIL_IMAGE }

    if ARCHIVE_FOLDER_ASSET != "":
        BODY["archiveFolderAsset"] = { "id": ARCHIVE_FOLDER_ASSET }

    # Set the appropriate fields based on the channel type
    if (TYPE == LIVE_CHANNEL_TYPES["External"]):
        BODY["externalUrl"] = URL

    try:
        # Send the request
        RESPONSE = requests.post(f"{ADMIN_URL}/liveChannel",  headers= HEADERS, data= json.dumps(BODY))
        INFO = json.loads(RESPONSE.text)

        if not RESPONSE.ok:
            raise Exception()
            
        wait_for_live_channel_status(AUTH_TOKEN, INFO["id"], LIVE_CHANNEL_STATUSES["Idle"], 120, 2)
        
        return json.loads(RESPONSE.text)

    except:
        api_exception_handler(RESPONSE, f"Created Live Channel {NAME} failed")

File: nomad-samples/admin/js/basic-live-channel/live-channel/create-live-channel.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";
import liveChannelStatuses from "./live-channel-statuses.js";
import waitForLiveChannelStatus from "./wait-live-channel-status.js";
import LIVE_CHANNEL_TYPES from "./live-channel-types.js";
import slugify from "../helpers/slugify.js";

export default async function createLiveChannel(AUTH_TOKEN, NAME, THUMBNAIL_IMAGE_ID, ARCHIVE_FOLDER_ASSET_ID,
                                                ENABLE_HIGH_AVAILABILITY, ENABLE_LIVE_CLIPPING, IS_SECURE_OUTPUT, 
                                                IS_OUTPUT_SCREENSHOTS, TYPE, URL) 
{
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);

    // Build the payload body
    const BODY = {
        name: NAME,
        routeName: slugify(NAME),
        enableHighAvailability: ENABLE_HIGH_AVAILABILITY,
        enableLiveClipping: ENABLE_LIVE_CLIPPING,
        isSecureOutput: IS_SECURE_OUTPUT,
        outputScreenshots: IS_OUTPUT_SCREENSHOTS,
        type: { id: LIVE_CHANNEL_TYPES[TYPE] }
    };

    
    if (THUMBNAIL_IMAGE_ID !== "")
    {
        BODY.thumbnailImage = { id: THUMBNAIL_IMAGE_ID };
    }

    if (ARCHIVE_FOLDER_ASSET_ID !== "")
    {
        BODY.archiveFolderAsset = { id: ARCHIVE_FOLDER_ASSET_ID };
    }

    // Set the appropriate fields based on the channel type
    if (TYPE === "External") {
        BODY.externalUrl = URL;
    }

    // Send the request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/liveChannel`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    });

    // Check for success
    if (RESPONSE && RESPONSE.ok) {
        // Parse JSON response
        const jsonResponse = await RESPONSE.json();

        // Wait for Live Channel to be idle if it was just created
        await waitForLiveChannelStatus(AUTH_TOKEN, jsonResponse.id, liveChannelStatuses.Idle, 120, 2);

        return jsonResponse;
    }

    await apiExceptionHandler(RESPONSE, `Creating Live Channel ${NAME} failed`);
}